### ISSUE TITLE
udev_monitor_filter_remove() returns -EINVAL

### ISSUE THREAD

**Issue by olysonek** [systemd/systemd#11529]
**systemd version the issue has been seen with**
239, systemd-239-8.gite339eae.fc29.x86_64

**Used distribution**
 Fedora

**Expected behaviour you didn't see**
 udev_monitor_filter_remove() should succeed.

**Unexpected behaviour you saw**
 udev_monitor_filter_remove() returns -EINVAL

**Steps to reproduce the problem**
```
$ cat a.c
#include <libudev.h>
#include <stdio.h>
#include <err.h>
#include <string.h>

int main(void)
{
	struct udev *ctx;
	struct udev_monitor *mon;
	int res;

	ctx = udev_new();
	if (ctx == NULL)
		errx(1, "udev_new\n");
	mon = udev_monitor_new_from_netlink(ctx, "udev");
	if (mon == NULL)
		errx(1, "udev_monitor_new_from_netlink");

	res = udev_monitor_filter_add_match_subsystem_devtype(mon, "disk", NULL);
	if (res < 0)
		errx(1, "udev_monitor_filter_add_match_subsystem_devtype: %s", strerror(-res));
	res = udev_monitor_filter_update(mon);
	if (res < 0)
		errx(1, "udev_monitor_filter_update: %s", strerror(-res));

	res = udev_monitor_filter_remove(mon);
	if (res < 0)
		errx(1, "udev_monitor_filter_remove: %s", strerror(-res));
	return 0;
}
$ gcc a.c -ludev
$ sudo ./a.out
a.out: udev_monitor_filter_remove: Invalid argument
```
If I'm reading the code right, this is because `(struct sock_fprog) { 0, NULL }` is passed to setsockopt:
https://github.com/systemd/systemd/blob/d1084aa2f803cdce6b3809dc9b7ec899251fd5dc/src/libsystemd/sd-device/device-monitor.c#L751

however that argument is treated as an invalid argument by the kernel:
https://github.com/torvalds/linux/blob/48b161983ae5266ffa42f0ccaf7224eaeda38e59/net/core/filter.c#L1007

**Comment by yuwata** [comment]
Fix is waiting in #11536.

**Comment by olysonek** [comment]
Wow, that was really quick. Thanks a lot everyone!
